### ISSUE TITLE
FFTW performance issue: how to override architecture-based choices?

### ISSUE THREAD

**Issue by RemiLacroix-IDRIS** [LLNL/spack#19998]
Hi,

We have an issue on our machine where AVX512-enabled FFTW builds are something like 3 times slower than AVX2 builds.

While trying to figure out this with the FFTW devs, we would like to override the automatic SIMD choice made by the Spack package based on the detected architecture (Intel Cascade Lake 6248).

I know I can force another arch when installing but it would create the corresponding module files in the wrong directory.

That would be the clever way to deal with this kind of issue?

Best regards,
Rémi

**Comment by haampie** [comment]
Maybe add a variant?

**Comment by RemiLacroix-IDRIS** [comment]
There was a variant for this but it got removed and replaced by auto-selection based on the architecture.

I could add it back with a default "auto" variant which would use the architecture.

**Comment by alalazo** [comment]
> I know I can force another arch when installing but it would create the corresponding module files in the wrong directory.

But you can use projections to force any naming you want, right?

**Comment by RemiLacroix-IDRIS** [comment]
> But you can use projections to force any naming you want, right?

No, the arch-based directory is always there (i.e you cannot project across arch-boundaries) which makes sense I guess.
